### PR TITLE
Fixes #4040 Isolate hard-coded js from member.php

### LINKED ISSUE #4040: Isolate hard-coded js from member.php
Isolate hard-coded registration form validator javascript chunks from `member.php` and transfer to a `.js` file.

### PR DESCRIPTION
**Pull request by effone**
Signed-off-by: effone <me@eff.one>

Resolves #4040

### CODE REVIEW

File: member.php [hunk 1]
```diff
@@ -1374,10 +1203,33 @@
...
+				maxnamelength: '{$mybb->settings['maxnamelength']}',
+				minpasswordlength: '{$mybb->settings['minpasswordlength']}',
+				captchaimage: '{$mybb->settings['captchaimage']}',
+				captchahtml: '{$captcha_html}',
+				securityquestion: '{$mybb->settings['securityquestion']}',
+				questionexists: '{$question_exists}',
+				requirecomplexpasswords: '{$mybb->settings['requirecomplexpasswords']}',
+				regtype: '{$mybb->settings['regtype']}',
+				hiddencaptchaimage: '{$mybb->settings['hiddencaptchaimage']}'
+			}
```
[the remark this answers is not in the document]
Comment (effone) [reply member.php]: Added.

**Comment by euantorano** [comment]
Great, thanks. Will merge when I’m back at the computer.

File: member.php [hunk 2]
```diff
@@ -1374,10 +1203,33 @@
...
+			};
+		
+			lang.js_validator_username_length = '{$lang->js_validator_username_length}';
+			lang.js_validator_invalid_email = '{$lang->js_validator_invalid_email}';
+			lang.js_validator_email_match = '{$lang->js_validator_email_match}';
+			lang.js_validator_not_empty = '{$lang->js_validator_not_empty}';
+			lang.js_validator_password_length = '{$lang->js_validator_password_length}';
+			lang.js_validator_password_matches = '{$lang->js_validator_password_matches}';
+			lang.js_validator_no_image_text = '{$lang->js_validator_no_image_text}';
+			lang.js_validator_no_security_question = '{$lang->js_validator_no_security_question}';
```
Comment (JordanMussi) [inline member.php:1231]:
```suggestion
			lang.js_validator_no_security_question = '{$lang->js_validator_no_security_question}';
			lang.js_validator_bad_password_security = '{$lang->js_validator_bad_password_security}';
```

File: jscripts/regvalidator.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+			minlength: regsettings.minpasswordlength,
+			equalTo: '#password',
+			messages: {
+				minlength: lang.js_validator_password_length,
+				required: lang.js_validator_password_length,
+				equalTo: lang.js_validator_password_matches
+			}
+		});
+	}
+});
```
Comment (JordanMussi) [inline jscripts/regvalidator.js:171]:
```suggestion
});

```

**Pull request merged**